extensions/2.0/Khronos/KHR_materials_variants/README.md

```diff
@@ -0,0 +1,148 @@
+
+In the second case, the shoe would have a white body and sole but red laces. In both cases, the meshes and their associated primitives remain the same with the exception of which material is used. Looking at the laces mesh, the KHR_materials_variants extension would define this mapping as follows:
+
+```javascript
+"meshes": [
+    {
+        "name": "laces",
+        "primitives": [
+            {
+                 ...,
+                "extensions": {
+                    "KHR_materials_variants" : {
+                        "mappings": [
+                            {
+                                "variants": [0],
+                                "material": 1,
+                            },
+                            {
+                                "variants": [1],
+                                "material": 2,
+                            },
+                        ],
+                    }
+                }
+            },
+            // ... more primitives ...
+        ]
+    },
+    // ... more meshes ...
+]
+```
+
+The variants property within each mapping refers to the index of a material variant located at the glTF root node:
+
+```javascript
+{
+  "asset": {"version": "2.0", "generator": "Fancy 3D Tool" },
+  "extensions": {
+      "KHR_materials_variants": {
+        "variants": [
+          {"name": "Red Shoe with White Laces" },
+          {"name": "White Shoe with Red Laces" },
+        ]
+      }
+  }
+}
+```
+
+## Interaction with existing glTF functionality
+
+_This section is non-normative._
+
+The primary purpose of this extension is to simply formalise the idea of static configurability, so that members of the ecosystem can meaningfully communicate: those who produce assets and those who ingest them, and all the tooling in between.
+
+As a secondary effect, material variants allow multiple assets — with shared geometry but different materials — to be stored more compactly, particularly relevant for the self-contained binary format GLB. When using external URIs as references to textures, applications may (optionally) process geometry only once and lazily request texture assets only when needed for a particular variant.
+
+## Implications for Applications and APIs
+
+_This section is non-normative._
+
+How does an application communicate to a glTF engine what the initial variant state should be? How does it submit a runtime request for a different configuration? It's out of scope for this extension to constrain or mandate an engine's public API, but a useful implementation will require something of the sort.
+
+## Examples
+
+_This section is non-normative._
+
+For reference, we provide several simple examples of variants, both represented as distinct constituent GLB files without the extension and as multi-variant GLB files leveraging this extension.  We also include the respective net size of the individual variant GLB files and the size of the single combined multi-variant GLB, for comparison.
```

Comment:
Would it better to remove this section entirely? I can see an issue with the sample or examples being moved elsewhere since they are in another repo and the spec document falls out of date.

Reply:
Yes, good point.  Many extensions include a small snippet of sample JSON for the extension itself, without including a complete working sample.  The samples do belong in a separate repo.

So, I think this section should be removed.